### ISSUE TITLE
Serialize horizontal coordinate instead of gridded operations

### ISSUE THREAD

**Issue by oliverwm1** [ai2cm/ace#523]
This would enable validation during inference that the provided initial condition and forcing data are on the same coordinate that was used to train the given stepper. Currently we only serialize the area weights which does not allow this (e.g. could have mismatched longitude coordinate while still having the same area weights).

See somewhat related discussion in ai2cm/ace#354. Possible we don't need the "GriddedOperations" class and we can just define these reduction operations directly on the horizontal coordinate classes?

**Comment by mcgibbon** [comment]
Good point. I would suggest implementing this by adding a `__eq__` operator or some method that checks for compatibility, and renaming the GriddedOperations class if needed. This requires exposing public methods that let one do this, since GriddedOperations is an ABC. You could think of this as being the same as defining the reduction operations directly on the horizontal coordinate classes.

HorizontalCoordinates has a significant number of methods defined that are specific to data loading and aren't needed where GriddedOperations is needed. I'd suggest we keep these decoupled by having separate classes used within and outside of data loading. There might be a way of doing this by separating the data loading specific concerns off to a new class, but I think the smallest change that doesn't merge these types together would be to add the ability for GriddedOperations to compare itself to another instance.

**Comment by oliverwm1** [comment]
I don't quite follow this proposal, since it complicates the `GriddedOperations` class which is otherwise quite simple.

If we just serialize the horizontal coordinate classes themselves, instead of the gridded ops, don't we get the best of both worlds in terms of keeping these things separate? We could still go from coordinate -> ops before passing along to the aggregator, corrector etc.

**Comment by mcgibbon** [comment]
Can we be explicit about where we're talking about? I thought the stepper, but it already serializes both the vertical coordinate and gridded operations, so I don't know what "instead of" means. Can't we already use this serialized attribute to validate inference data?

**Comment by oliverwm1** [comment]
> Can we be explicit about where we're talking about? I thought the stepper, but it already serializes both the vertical coordinate and gridded operations, so I don't know what "instead of" means. Can't we already use this serialized attribute to validate inference data?

Yes, I am talking about the stepper. We cannot use the existing serialized gridded operations attribute to validate inference data because we only serialize the area weights, not the coordinates themselves. I mentioned this in the issue above:

> Currently we only serialize the area weights which does not allow this (e.g. could have mismatched longitude coordinate while still having the same area weights).

**Comment by mcgibbon** [comment]
Ok but can we use the existing serialized vertical_coordinates attribute? Has this issue been resolved?

**Comment by oliverwm1** [comment]
Yes, we can validate the vertical coordinate with what we currently serialize. But we cannot validate the horizontal coordinate so the issue is still outstanding.

**Comment by mcgibbon** [comment]
Ahh, I see. Thanks for clarifying.
